pkg/util/dynamicobject/dynamicobject.go

```diff
@@ -23,20 +23,20 @@ package dynamicobject
 // should not be `map[interface{}]interface{}`.
 type DynamicObject map[string]interface{}
 
-// UnmarshalYAML implements yaml.Unmarshaler
+// UnmarshalJSON implements json.Unmarshaler
```

Comment:
the changes in this file are incorrect. `UnmarshalYAML` exists to convert `map[interface{}]interface{}` to `map[string]interface{}`, this is not needed for JSON.

Reply:
Oops, so if we adopt it in json format, can we just delete this method? As you said, json does not have this problem.

Reply:
propose to revert the changes, the function will only be called for YAML, this makes the behavior correct for both cases.

Reply:
Updated.